git-cms-checkdeps

```diff
@@ -1,288 +1,293 @@
+    if match:
+        pack = match.group(2)
+        if pack not in upackages:
+            upackages[pack] = []
+            if not pack:
+                print(">> Package removed $pack")
+        upackages[pack].append(diff)
+
+packages = sorted(upackages)
+for package in packages:
+    if re.search(r'^UserCode.*', package):
+        continue
+    print(">> Checking %s %s" % (package, reltag))
+    msgs = {"-": {}, "x": {}}
+    for diff in upackages[package]:
+        _del = None
+        match = re.search(r'^(R[0-9]*)(\s+[^\s]+)\s+', diff)
+        if match:
+            diff = "D %s" % match.group(2)
+        match_M = re.search(r'^(M)', diff)
+        match_U = re.search(r'^(U)', diff)
+        match_D = re.search(r'^(D)', diff)
+        if match_M or match_U or match_D:
+            for match in [match_M, match_U, match_D]:
+                if match:
+                    if match.group() == "D":
+                        _del = 1
+                    sp2 = diff.split(' ')
+                    diff = sp2[1]
+        else:
+            diff = None
+        if diff:
+            msgs["x"][diff] = 1
+        if _del:
+            msgs["-"][diff] = 1
+    files = sorted(msgs["-"].keys())
+    deletedFiles.extend(files)
+    changedFiles.extend(files)
+    for diff in files:
+        print("   - %s" % diff)
+        del msgs["x"][diff]
+    files = sorted(msgs["x"].keys())
+    changedFiles.extend(files)
+    for diff in files:
+        print("   x %s" % diff)
+
+if poison_includes:
+    poisonIncludes(deletedFiles, localtop)
+if poison_only:
+    sys.exit(0)
+
+recompileList = {}
+big_bf = ""
+for file in changedFiles:
+    if re.search(r'^BigProducts\/[^\/]+\/BuildFile\.xml$', file):
+        big_bf = "%s %s" % (big_bf, file)
+    if file not in vals:
+        continue
+    for dep in vals[file]:
+        sp = dep.split("/")
+        try:
+            recompile = "%s/%s" % (sp[0], sp[1])
+        except IndexError:
+            pass
+        if not list(filter(lambda x: recompile in x, packages)):
+            scope = "header"
+            if re.search(r'\.py$', file):
+                scope = "python"
+            elif re.search(r'\/BuildFile(\.xml|)$', file):
+                scope = "buildfile"
+            if recompile not in recompileList:
+                recompileList[recompile] = {}
+            if scope not in recompileList[recompile]:
+                recompileList[recompile][scope] = {}
+            recompileList[recompile][scope] = file
+
+if big_bf != "":
+    pipe_0 = subprocess.Popen(["git diff -r %s -- %s" % (
+            reltag, big_bf)], stdout=subprocess.PIPE)
+    pipe_1 = subprocess.Popen(["sed -e 's/[ \t]\+/ /"], stdin=pipe_0.stdout, stdout=subprocess.PIPE)
+    pipe_2 = subprocess.Popen(["grep '^+'"], stdin=pipe_1.stdout, stdout=subprocess.PIPE)
+    pipe_3 = subprocess.Popen(["grep -i '< *use  *name *='"], stdin=pipe_2.stdout, stdout=subprocess.PIPE)
```

Comment:
all these Popen are not going to work, you need list of items not a single item in the list.
Please use only one pipe_0 and rest do it in python. rest is just search for '< *use +name *='